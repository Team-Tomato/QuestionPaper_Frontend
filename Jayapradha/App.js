import React, { Component } from 'react';
import './App.css';
import './style.css';
class App extends Component {
  send()
  {
    alert("Thank you for your response.We will contact you as soon as possible!!")
  }
  render(){
  return (
    <div>
      <div className="header">
        <h1>We wish to hear from you!</h1>
        <p className="sansserif" style={{color:"#D3D3D3"}}>Have a question?Get in touch with us.We're here for you</p>
       </div>
        
        <div className="box" style={{padding:"100px"}}>
        <h2 className="serif" style={{color:"#000080"}}>Let's get Together</h2>
          <h4 className="sansserif" style={{color:"#E75480"}} > Stop by and say Hi to teamtomato.oss@gmail.com</h4>
          </div>

          <div className="box" >
            <form style={{padding:"5px"}}>
              
            <h2 className="serif" style={{color:"#654321"}}>Give us a shout!!</h2>
            <h4 className="sanserif" style={{color:"A9A9A9"}}>Use the quick form below to drop us a message</h4>
            
            <div className="center">
              
            <div style={{textAlign:"left"}}>
            <label className="sanserif" style={{fontWeight:"bold"}}>Name</label><br/><br/>
            <input type='text' size="28"/><br/><br/>
            <label className="sanserif" style={{fontWeight:"bold"}}>Email</label><br/><br/>
            <input type='text' size="28"/><br/><br/>
            
            <label className="sanserif" style={{fontWeight:"bold"}}>Message</label><br/><br/>
            <textarea rows="5" cols="30"/><br/><br/>
            <button className="buttonstyle"  onClick={this.send}>Send</button>
            </div>
            </div>
            </form>
            </div>
            
     </div> 
      
  );}
}

export default App;
