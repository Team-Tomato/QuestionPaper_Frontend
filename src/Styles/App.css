  .container{
    padding:20px;
  }

  .row-align{
    justify-content: center;
  }

  ::placeholder {
    text-align: center;
    font-family: "Times New Roman", Times, serif;
  }