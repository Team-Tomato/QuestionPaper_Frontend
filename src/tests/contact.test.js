import React from 'react';
import App from '../Views/contact';
import {create} from 'react-test-renderer';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import { Button } from 'reactstrap';
import '../setupTest.js';

describe("Form Testing",()=>{
    test ("Matches the snapshot",()=>{
        const form = create(<App />);
        expect(form.toJSON()).toMatchSnapshot();        
    });
    
    test("calls sent method when Send button is clicked",()=>{
        const submitHandler = jest.fn();
        const wrapper = mount (<Button onClick = {submitHandler} />);
        const form = wrapper.find('Button');
        form.simulate('click');
        expect(submitHandler.mock.calls.length).toEqual(1);
    });    

    test('input check',()=>{
        let wrapper;
        window.alert = jest.fn();
        wrapper = shallow(<App />);
        wrapper.find('Input[type="text"]').simulate('change',{target:{name:'name',value:'testing'}});
        wrapper.find('Input[type="email"]').simulate('change',{target:{name:'email',value:'testing@gmail.com'}});
        wrapper.find('Input[type="text"]').simulate('change',{target:{name:'msg',value:'testing'}});
        expect(wrapper.state('name')).toEqual('testing');
        expect(wrapper.state('email')).toEqual('testing@gmail.com');
        expect(wrapper.state('msg')).toEqual('testing');
        wrapper.find('Button').simulate('click');
        expect(window.alert.mock.calls.length).toBe(1);       
    });

    });    








