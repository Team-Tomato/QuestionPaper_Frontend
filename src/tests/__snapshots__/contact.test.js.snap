exports[`Form Testing Matches the snapshot 1`] = `
<div>
  <div
    className="header">
    <h1>
      We will be happy to hear from you !
    </h1>
    <h4>
      Get in touch with us.
    </h4>
  </div>
  <div
    className="container">
    <div
      className="indentCorrection card">
      <div
        className="card-body">
        <h5>
          Use the public key
        </h5>
        <h2
          className="violet">
          D16C 2228 B32C EA4A B6B8 B92B EE4D DD8F DA96 7765
        </h2>
        <h5>
          to mail us at teamtomato.oss@gmail.com
        </h5>
      </div>
    </div>
    <h3
      className="center">
      or
    </h3>
    <form
      className="">
      <input
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="Your Name"
        sm={10}
        type="text"
        value="" />
      <br />
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        placeholder="youremail@email.com"
        sm={10}
        type="email"
        value="" />
      <br />
      <textarea
        className="form-control"
        name="message"
        onChange={[Function]}
        placeholder="Your Message"
        sm={10}
        value="" />
      <br />
      <button
        aria-label={null}
        className="btn btn-secondary btn-block"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "violet",
          }
        }
        type="button"
        variant="contained">
        Send
      </button>
    </form>
  </div>
</div>
`;
